numpy/random/mtrand/randomkit.c

```diff
@@ -233,33 +233,33 @@ rk_random(rk_state *state)
 
     /* Tempering */
     y ^= (y >> 11);
-    y ^= (y << 7) & 0x9d2c5680UL;
-    y ^= (y << 15) & 0xefc60000UL;
+    y ^= (y << 7) & 0x9d2c5680ULL;
+    y ^= (y << 15) & 0xefc60000ULL;
     y ^= (y >> 18);
 
     return y;
 }
 
-long
+long long
 rk_long(rk_state *state)
 {
     return rk_ulong(state) >> 1;
 }
 
-unsigned long
+unsigned long long
 rk_ulong(rk_state *state)
 {
-#if ULONG_MAX <= 0xffffffffUL
+#if ULLONG_MAX <= 0xffffffffULL
     return rk_random(state);
 #else
     return (rk_random(state) << 32) | (rk_random(state));
```

Comment:
To get this part to work with long long (after reverting the change of datatype in the rng), you need a cast: `((long long)rk_random(state) << 32) | ...`